packages/terra-functional-testing/src/services/wdio-visual-regression-service/VisualRegressionLauncher.js

```diff
@@ -63,7 +64,30 @@ class VisualRegressionLauncher {
    * Function to be executed after a test in Mocha.
    * @param {Object} test - test details
    */
-  afterTest() {
+  afterTest(test, _, { passed }) {
+    const { config } = browser;
+    const { formFactor } = config.launcherOptions;
+    const screenshotContext = {
+      desiredCapabilities: this.context.desiredCapabilities,
+      suite: this.currentSuite,
+      test: this.currentTest,
+      meta: {
+        currentFormFactor: formFactor,
+      },
+      options: {
+        name: test.title,
+      },
+    };
+
+    const screenshotContextCleaned = lodashPickby(screenshotContext, lodashIdentity);
+    const { errorPath } = this.compare.getScreenshotPaths(screenshotContextCleaned);
+    if (passed) {
+      fse.removeSync(errorPath);
+    } else {
+      fse.ensureFileSync(errorPath);
+      browser.saveScreenshot(errorPath);
```

Comment:
It was the whole document before. It also taking the screenshot for every test that fails not just the ones that we do a screenshot comparison for, if a selector is provided for screenshot comparison, it may not be relevant for why the test is failing, it could be failing due to other assertions in the test.